pkg/controller-gen/main.go

```diff
@@ -265,6 +276,56 @@ func generateClientset(groups map[string]bool, customArgs *cgargs.CustomArgs) er
 	)
 }
 
+func generateOpenAPI(groups map[string]bool, customArgs *cgargs.CustomArgs) error {
```

Comment:
While trying out the changes here for types in rancher/rancher, I couldn't get it to generate anything.

The generator works only when the directive `// +k8s:openapi-gen=true` is in the `doc.go` file. Adding it manually won't work because another code generator overwrites this doc.go file. This can be added here: https://github.com/rancher/wrangler/blob/241bfe6cb00df18d82e08966e8eb0217461cec39/pkg/controller-gen/generators/client_generator.go#L96-L100.

Note: This ends up in many files which we don't need (doc.go, zz_generated_register.go, zz_generated_list_types.go). Not ideal, but it's already like this and we want to get rid of gengo anyway.

Reply:
Just added! Thank you!